lambda/src/main/resources/db/migration/V15__add-consignment-seq-and-reference.sql

```diff
@@ -0,0 +1,15 @@
+-- Create sequence that will be the consignment sequence ID used to encode b25 section of consignment reference
+CREATE SEQUENCE "ConsignmentSequenceID"
+NO CYCLE
+START WITH 1
+INCREMENT BY 1;
+
+-- Alter Consignment table to add column to store sequence ID and another for full consignment reference
+ALTER TABLE "Consignment"
+ADD COLUMN "ConsignmentSequence" bigint DEFAULT nextval('"ConsignmentSequenceID"'),
+ADD COLUMN "ConsignmentReference" text,
+ADD UNIQUE ("ConsignmentSequence", "ConsignmentReference");
+
+-- Alter consignment table to create consignment reference for existing Consignment rows
+UPDATE "Consignment"
+SET "ConsignmentReference"='TDR-2020-OLD-' || "ConsignmentSequence";
```

Comment:
How does `'TDR-2020-OLD-' || "ConsignmentSequence"` work? I'm not familiar with the syntax - the `||` looks like a boolean `OR`, but it seems to be doing string concatenation?

Reply:
Yeah, it's PostgreSQL's concatenation operator. There is a `CONCAT()` function too, but I couldn't get it to work properly within the file. I'll give `CONCAT()` another go, and if I can't get it to work I can add a comment to explain that it's concatenation rather than `OR`